kinto/core/cache/postgresql/__init__.py

```diff
@@ -8,6 +9,17 @@
 from kinto.core.utils import json
 
 
+def retry_on_failure(func):
+    @wraps(func)
+    def wraps_func(self, *args, **kwargs):
+        import psycopg2
+        try:
+            return func(self, *args, **kwargs)
+        except psycopg2.IntegrityError:
+            return func(self, *args, **kwargs)
```

Comment:
We could retry 10 times with a small increasing delay between each call after the second call